src/Acuminator/Acuminator.Analyzers/StaticAnalysis/LegacyBqlField/LegacyBqlFieldAnalyzer.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Acuminator.Analyzers.StaticAnalysis.LegacyBqlField
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class LegacyBqlFieldAnalyzer : PXDiagnosticAnalyzer
+	{
+		public const string CorrespondingPropertyType = "CorrespondingPropertyType";
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Descriptors.PX1060_LegacyBqlField);
+
+		protected override bool ShouldAnalyze(PXContext pxContext) => pxContext.IsAcumatica2019R1;
+
+		internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+		{
+			compilationStartContext.RegisterSymbolAction(
+				c => Analyze(c, pxContext),
+				SymbolKind.NamedType);
+		}
+
+		private async void Analyze(SymbolAnalysisContext context, PXContext pxContext)
+		{
+			if (context.Symbol is INamedTypeSymbol dacFieldType)
+			{
+				if (!IsDacFieldType(dacFieldType, pxContext) || AlreadyStronglyTyped(dacFieldType, pxContext))
```

Comment:
@Derrekam this diagnostic should run for DACs and analyze all fields of a DAC/ DAC extension. It will be far more perfomant than analyze each fialed of the same DAC separately and perform all required checks. Moreover, if you analyze whole DAC you can remove some of the checks in `IsDacFieldType` method like:
```
dacFieldType.ContainingType != null &&
dacFieldType.ContainingType.IsDacOrExtension(pxContext);
```

Reply:
And how many fixes should "mass" diagnostic provide? One for a whole DAC, or many for every single field?

Reply:
@Derrekam  I think, it depends on your vision of your feature. The more detailed approach is to register code fix for every DAC field, but on the other side it will be strange to make fix for one DAC field and don't make it for another. 
Still it is good to show warning not on the DAC name but on the specific field.

Reply:
@Derrekam  I have sent my thoughts about possible implementations to you in Slack